sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -235,9 +237,28 @@ case class Union(children: Seq[LogicalPlan]) extends LogicalPlan {
   }
 
   // updating nullability to make all the children consistent
-  override def output: Seq[Attribute] =
-    children.map(_.output).transpose.map(attrs =>
-      attrs.head.withNullability(attrs.exists(_.nullable)))
+  override def output: Seq[Attribute] = {
+    children.map(_.output).transpose.map { attrs =>
+      val firstAttr = attrs.head
+      val nullable = attrs.exists(_.nullable)
+      try {
```

Comment:
this is a minor bug fix and I think no backport is needed. This try-cache looks overkill now.

Reply:
Based on https://issues.apache.org/jira/browse/SPARK-26812 it sounds like this can have a correctness impact for queries and it looks like a pretty straightforward fix. Given this, I think we should consider a 2.4.x backport.

Reply:
sounds reasonable. @viirya can you send a new PR for branch 2.4? thanks!

Reply:
@cloud-fan I can, but if you really mention me not @mgaido91?

Reply:
ah sorry, my mistake :P

@mgaido91 can you send a backport PR please?

Reply:
+1 for backporting this!

Reply:
sure,I am doing it, thanks